mh/arena_utils.py

```diff
@@ -0,0 +1,200 @@
+"""
+Arena Quest ID List:
+EA60 
+EA61
+EA62
+EA63
+EA64
+EA65
+EA66
+EA67
+EA68
+EA69
+EA6A
+EA6B
+"""
+
+# bird and brute bowgun: rathling gun + barrel, poison stinger frame, light bowgun stock
+# sea power bowgun: rathling gun barrel, rathling gun + frame, rathling gun + stock
+# land lords bowgun: jho barrel, diablos frame, barioth stock
+# two flames bowgun: lagiacrus barrel, lagiacrus frame, barioth stock
+
+import csv
+from quest_utils import make_binary_event_quest,\
+    generate_flags, Monster, LocationType,\
+    QuestRankType, QuestRestrictionType, ResourcesType,\
+    StartingPositionType, ItemsType
+from equipment_utils import Chestpiece, Gauntlets, Faulds,\
+    Leggings, Helmet, EquipmentClasses, Greatsword,\
+    SnS, Hammer, Longsword, Switchaxe, Lance,\
+    BowgunFrame, BowgunStock, BowgunBarrel,\
+    slot, item_slot
+
+
+def create_arena_equipment_set(GRUDGE_MATCH, sets):
+    for set_num in range(4):
+        weapon1, weapon2, weapon3, helm, chestpiece, gauntlets,\
+            faulds, leggings, items, gunner_pouch = sets[set_num]
+        # ---------- SET 1 ----------
+        # Weapon
+        GRUDGE_MATCH += slot(*weapon1)
+        if weapon2 is not None:
+            GRUDGE_MATCH += slot(*weapon2)
+        else:
+            GRUDGE_MATCH += b"\xff" * 0x04
+        if weapon3 is not None:
+            GRUDGE_MATCH += slot(*weapon3)
+        else:
+            GRUDGE_MATCH += b"\xff" * 0x04
+        # Armors
+        GRUDGE_MATCH += slot(EquipmentClasses.Helmet, helm)
+        GRUDGE_MATCH += slot(EquipmentClasses.Chestpiece, chestpiece)
+        GRUDGE_MATCH += slot(EquipmentClasses.Faulds, faulds)
+        GRUDGE_MATCH += slot(EquipmentClasses.Gauntlets, gauntlets)
+        GRUDGE_MATCH += slot(EquipmentClasses.Leggings, leggings)
```

Comment:
Ditto. This would not save any space or make it any more easy to read. We would be turning

```
GRUDGE_MATCH += slot(EquipmentClasses.Helmet, helm)
GRUDGE_MATCH += slot(EquipmentClasses.Chestpiece, chestpiece)
GRUDGE_MATCH += slot(EquipmentClasses.Faulds, faulds)
GRUDGE_MATCH += slot(EquipmentClasses.Gauntlets, gauntlets)
GRUDGE_MATCH += slot(EquipmentClasses.Leggings, leggings)
```

into

```
def pack_armor(helm, chestpiece, faulds, gauntlets, leggings):
        GRUDGE_MATCH += slot(EquipmentClasses.Helmet, helm)
        GRUDGE_MATCH += slot(EquipmentClasses.Chestpiece, chestpiece)
        GRUDGE_MATCH += slot(EquipmentClasses.Faulds, faulds)
        GRUDGE_MATCH += slot(EquipmentClasses.Gauntlets, gauntlets)
        GRUDGE_MATCH += slot(EquipmentClasses.Leggings, leggings)

pack_armor(helm, chestpiece, faulds, gauntlets, leggings)
```

Reply:
That's not the point, it's to make the code self-contained and clearer, you're also missing the return value in your `pack_armor` example.

Reply:
There wouldnt need to be a return value if the function is defined on the same level as `pack_armor` would be called, it would modify the `GRUDGE_MATCH` within the parent function's scope.

Reply:
That makes the function less obvious and useless, then. Moreover, you're missing the the `global` keyword to do such modification. Again, keep things simple and don't rely on features you aren't comfortable with. What I meant was something simpler such as:
```python
def pack_armor(helm, chestpiece, faulds, gauntlets, leggings):
    data = slot(EquipmentClasses.Helmet, helm)
    data += slot(EquipmentClasses.Chestpiece, chestpiece)
    data += slot(EquipmentClasses.Faulds, faulds)
    data += slot(EquipmentClasses.Gauntlets, gauntlets)
    data += slot(EquipmentClasses.Leggings, leggings)
    return data

grudge_match += pack_armor(helm, chestpiece, faulds, gauntlets, leggings)
```